File: Wirtualny Swiat/PiotrPrzymus193736JavaWS/Wilk.java
package WSJAVA;

public class Wilk extends Zwierze {
    public Wilk(int polX, int polY, Swiat swiat) {
        super(polX, polY, swiat);
        this.sila = 9;
        this.inicjatywa = 5;
        this.znak_gatunku = Stale.WILK;
        this.wiek = 0;
        this.nazwa = "Wilk";
    }
    public Wilk(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak_gatunku, int wiek, String nazwa, Swiat swiat) {
        super(polozenieX, polozenieY, sila, inicjatywa, znak_gatunku, wiek, nazwa, swiat);
    }

    @Override
    public void PostawDziecko(int x, int y)
    {
        swiat.DodajOrganizm(new Wilk(x, y, swiat));
        //swiat.gotoxy(40, 8);
    }
}

File: Wirtualny Swiat/Piotr Przymus 193736 Wirtualny Swiat/Wilk.cpp
#include "Wilk.h"

Wilk::Wilk(int polX, int polY, Swiat& swiat)
	: Zwierze(polX, polY, swiat)
{
	wiek = 0;
	inicjatywa = 5;
	sila = 9;
	znak_gatunku = WILK;
	nazwa = "Wilk";
}

Wilk::Wilk(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak, int wiek, string nazwa, Swiat& swiat)
	: Zwierze(polozenieX, polozenieY, sila, inicjatywa, znak, wiek, nazwa, swiat)
{
}

Wilk::~Wilk()
{

}

void Wilk::PostawDziecko(int x, int y)
{
	swiat.DodajOrganizm(new Wilk(x, y, swiat));
	swiat.gotoxy(40, 8);
}